third_party/move/move-compiler-v2/transactional-tests/tests/no-v1-comparison/closures/persistent.move

```diff
@@ -0,0 +1,24 @@
+//# publish
+module 0x66::test {
+    use 0x1::signer;
+
+    struct Work(||u64) has store, key;
+
+    #[persistent]
```

Comment:
1. Should we also have a compiler check that these new attributes should not be used on functions within scripts?
2. Should we add a lint rule (not in this PR) that using `#[persistent]` with `public` (and even `entry`) functions is redundant (as they are already persistent)?

Reply:
Done